mc/Diameter.package/AVPPathTest.class/class/s6aParser.st

```diff
@@ -0,0 +1,11 @@
+accessing
+s6aParser
+	^ OTPDiaParser new
+		moduleSet:
+			(Dictionary new
+				at: 'ietf-avps' put: 'contrib/specs/ietf-avps.dia';
```

Comment:
This works for Pharo7 CI but not Pharo6. For TCAP we added this TCAPModuleLibrary concept and inlined the text. Is this an option here?

Reply:
I noticed! 
I've kept them in the original files to make it explicit when they change version but the CI got nervous :D 
Inlining those definitions in a class should do and get versioned too. Great feedback, thanks!

Reply:
@zecke When trying to address this I found that `OTPDiaParser>>inherits` is coupled with the idea of having paths in its `moduleSet`  dictionary and read them from it when parsing (hence coupled with some kind of path). 

If we decouple it from this idea and make it deal with just a given source, then we could make that source to come from accessing from wherever, for example an `S6aLibrary` as you suggested. I was trying to do that but when I found this I got afraid of side effects. 

My thinking is to have instead of
```parser moduleSet:
            (Dictionary new
                at: 'ietf-avps'
                    put: 'src/diameter/contrib/specs/ietf-avps.dia'```

something that puts just the contents there so you could have the same output when you instantiate it by:
```parser moduleSet:
            (Dictionary new
                at: 'ietf-avps'
                    put: 'src/diameter/contrib/specs/ietf-avps.dia' asFileReference contents```

than by doing:

```parser moduleSet:
            (Dictionary new
                at: 'ietf-avps'
                    put: S6aLibrary specIetfAvps```

Can you shred some light on this? What would you suggest? 

This is the problematic code (note the ` asFileReference contents` part):
```
	^super inherits ==> [:p |
		self flag: #todo. "Handle p third as a type filter. Not sure how that works"
		OTPDiaInheritToken new
			moduleName: p second;
			parsed: (self parse: (moduleSet at: p second) asFileReference contents)].```

Reply:
Let's just do the asFileReference contents early and remove the assumption from the parser.

* Loading all diameter files ever written.. fits into 640kb RAM so we don't need to optimize
* If it wouldn't then loading them into the image would fail..
* We can optimize by having an InMemoryFile.. but as above we don't have a need to optimize.

Reply:
Good point!